.github/workflows/build-images.yaml

```diff
@@ -0,0 +1,223 @@
+  IMG_SCHEDULER:          "neondatabase/autoscale-scheduler"
+  IMG_AUTOSCALER_AGENT:   "neondatabase/autoscaler-agent"
+  IMG_CLUSTER_AUTOSCALER: "neondatabase/cluster-autoscaler-neonvm"
+
+defaults:
+  run:
+    shell: bash -euo pipefail {0}
+
+jobs:
+  tags:
+    outputs:
+      controller:         ${{ steps.show-tags.outputs.controller }}
+      vxlan-controller:   ${{ steps.show-tags.outputs.vxlan-controller }}
+      runner:             ${{ steps.show-tags.outputs.runner }}
+      scheduler:          ${{ steps.show-tags.outputs.scheduler }}
+      autoscaler-agent:   ${{ steps.show-tags.outputs.autoscaler-agent }}
+      cluster-autoscaler: ${{ steps.show-tags.outputs.cluster-autoscaler }}
+    runs-on: [ self-hosted, gen3, small ]
+    steps:
+      - id: show-tags
+        run: |
+          echo "controller=${{ env.IMG_CONTROLLER }}:${{ inputs.tag }}" | tee -a $GITHUB_OUTPUT
+          echo "vxlan-controller=${{ env.IMG_VXLAN_CONTROLLER }}:${{ inputs.tag }}" | tee -a $GITHUB_OUTPUT
+          echo "runner=${{ env.IMG_RUNNER }}:${{ inputs.tag }}" | tee -a $GITHUB_OUTPUT
+          echo "scheduler=${{ env.IMG_SCHEDULER }}:${{ inputs.tag }}" | tee -a $GITHUB_OUTPUT
+          echo "autoscaler-agent=${{ env.IMG_AUTOSCALER_AGENT }}:${{ inputs.tag }}" | tee -a $GITHUB_OUTPUT
+          echo "cluster-autoscaler=${{ env.IMG_CLUSTER_AUTOSCALER }}:${{ inputs.tag }}" | tee -a $GITHUB_OUTPUT
+
+  vm-kernel:
+    if: ${{ ! inputs.skip }}
+    uses: ./.github/workflows/vm-kernel.yaml
+    with:
+      tag: ${{ inputs.kernel-image || inputs.tag }}
+      return-image-for-tag: ${{ inputs.kernel-image }}
+    secrets: inherit
+
+  build:
+    if: ${{ ! inputs.skip }}
+    needs: [ tags, vm-kernel ]
+    runs-on: [ self-hosted, gen3, large ]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # fetch all, so that we also include tags
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+          # Disable cache on self-hosted runners to avoid /usr/bin/tar errors, see https://github.com/actions/setup-go/issues/403
+          cache: false
+        # Sometimes setup-go gets stuck. Without this, it'll keep going until the job gets killed
+        timeout-minutes: 10
+
+      # Use 'git describe' for embedding git information (duplicated from the Makefile)
+      - name: get git info
+        id: get-git-info
+        run: |
+          # note: --tags enables matching on lightweight (i.e. not annotated) tags, which normally
+          # wouldn't be necessary, except that actions/checkout@v3 does weird things to setup the
+          # repository that means that we actually end up checked out with *just* a lightweight tag
+          # to the tagged commit.
+          echo "info=$(git describe --tags --long --dirty)" >> $GITHUB_OUTPUT
+
+      - name: get CA base git tag
+        id: get-ca-tag
+        if: ${{ inputs.build-cluster-autoscaler }}
```

Comment:
Oh, _yikes_. I think there's a number of other places that need to change because of that :grimacing:.

Tried to do that in df5deae0a1ad7e41cd8cc28de6b917c38c206ec5 — does that match your understanding?

Reply:
Yep, that's exactly what I meant.

I'd love `actionlint` to catch such cases.